stingray/lightcurve.py

```diff
@@ -408,16 +408,15 @@ def check_lightcurve(self):
 
         check_gtis(self.gti)
 
-        dt_array = []
-        for g in self.gti:
-            mask = create_gti_mask(self.time, [g], dt=self.dt)
-            t = self.time[mask]
-            dt_array.extend(np.diff(t))
-        dt_array = np.asarray(dt_array)
-
-        if not (np.allclose(dt_array, np.repeat(self.dt, dt_array.shape[0]))):
-            simon("Bin sizes in input time array aren't equal throughout! "
-                  "This could cause problems with Fourier transforms. "
+        idxs = np.searchsorted(self.time, self.gti)
+        time_diff = np.empty([1,])
+
+        for idx in range(idxs.size - 1):
+            np.append(time_diff, np.diff(self.time[idxs[idx][0]:idxs[idx][1]]))
+
+        if not (np.allclose(time_diff, np.repeat(self.dt, time_diff.shape[0]))):
+            simon("Bin sizes in input time array aren't equal throughout!"
+                  "This could cause problems with Fourier transforms."
```

Comment:
I think this can be changed to
```suggestion
        uneven = False
        for idx in range(idxs.size - 1):
            local_diff = np.diff(self.time[idxs[idx][0]:idxs[idx][1]])
            if np.any(~np.isclose(local_diff, self.dt)):
                uneven = True
                break
        if uneven:
            simon("Bin sizes in input time array aren't equal throughout!"
                  "This could cause problems with Fourier transforms."
```

Reply:
Yes, it will be wayy faster than my implementation!